### PR TITLE
Convenience function exposing available panel types.

### PR DESCRIPTION
**Pull request by kevinrue**

### CODE REVIEW

File: R/constants.R [hunk 1]
```diff
@@ -290,7 +289,7 @@ all_panel_types <- c(point_plot_types, linked_table_types, custom_panel_types, "
 #' \code{\link{.decode_panel_name}}
 .encode_panel_name <- function(names) {
     id <- as.integer(gsub(".* ([0-9]+)$", "\\1", names))
-    raw.str <- rev.translation[gsub(" [0-9]+$", "", names)]
+    raw.str <- availablePanelTypes(row.names="decoded")[gsub(" [0-9]+$", "", names), "encoded", drop=TRUE]
```
Comment (LTLA) [inline R/constants.R]: This is pretty damn complicated.

File: R/misc.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#'
+#' @examples
+#' availablePanelTypes()
+availablePanelTypes <- function(row.names=c(NA, "encoded", "decoded")){
+    
+    row.names <- match.arg(row.names)
+    
+    df <- data.frame(
+        decoded=panelTypes,
+        encoded=names(panelTypes),
```
Comment (LTLA) [inline R/misc.R]: I don't see that this is necessary, the user should never see the encoded panel names AFAIK.

Comment (kevinrue) [reply R/misc.R]: Check out the Rd page. I was thinking about users/developers who want to write tours. Knowing the encoded panel names becomes necessary there.

File: R/constants.R (hunk 1 above)
Comment (kevinrue) [reply R/constants.R]: True.
Though I was a bit bothered by the need for both `translation` and `rev.translation`.

File: R/misc.R (hunk 2 above)
Comment (LTLA) [reply R/misc.R]: Can't we just return `panelTypes`, rather than a full fledged function?

File: R/constants.R (hunk 1 above)
Comment (LTLA) [reply R/constants.R]: Well, it's definitely easier to understand than this.

Comment (kevinrue) [reply R/constants.R]: See b37d912e660ac9a7ff2f9c84b8f2fcd4bd00d172 for my new offer.

File: R/misc.R (hunk 2 above)
Comment (kevinrue) [reply R/misc.R]: I've implemented your suggestion in b37d912e660ac9a7ff2f9c84b8f2fcd4bd00d172

File: R/iSEE-main.R [hunk 3]
```diff
@@ -271,7 +272,7 @@ iSEE <- function(se,
 
         dashboardSidebar(
             selectizeInput("newPanelChoice", label="Choose panel type:",
-                selected=rev.translation[1], choices=rev.translation),
+                selected=names(panelTypes)[1], choices=structure(names(panelTypes), names=panelTypes)),
```
Comment (LTLA) [inline R/iSEE-main.R]: And `panelCodes` here, as well?

**Pull request merged**